code/game/objects/effects/anomalies.dm

```diff
@@ -1,213 +1,350 @@
+					var/mob/M = A
+					if(M.client)
+						INVOKE_ASYNC(src, .proc/blue_effect, M)
+
+/obj/effect/anomaly/bluespace/proc/blue_effect(mob/M)
+	var/obj/blueeffect = new /obj(src)
+	blueeffect.screen_loc = "WEST,SOUTH to EAST,NORTH"
+	blueeffect.icon = 'icons/effects/effects.dmi'
+	blueeffect.icon_state = "shieldsparkles"
+	blueeffect.layer = FLASH_LAYER
+	blueeffect.plane = FULLSCREEN_PLANE
+	blueeffect.mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	M.client.screen += blueeffect
+	sleep(20)
+	M.client.screen -= blueeffect
+	qdel(blueeffect)
 
-/obj/effect/anomaly/bluespace/Bumped(atom/A)
-	if(isliving(A))
-		do_teleport(A, locate(A.x, A.y, A.z), 10)
-	return
 
 /////////////////////
 
 /obj/effect/anomaly/pyro
 	name = "pyroclastic anomaly"
 	icon_state = "mustard"
-
-/obj/effect/anomaly/pyro/New()
-	..()
-	aSignal.origin_tech = "plasmatech=7"
+	var/ticks = 0
+	aSignal = /obj/item/assembly/signaler/anomaly/pyro
 
 /obj/effect/anomaly/pyro/anomalyEffect()
 	..()
+	ticks++
+	if(ticks < 5)
+		return
+	else
+		ticks = 0
 	var/turf/simulated/T = get_turf(src)
 	if(istype(T))
-		T.atmos_spawn_air(LINDA_SPAWN_HEAT | LINDA_SPAWN_TOXINS, 3)
+		T.atmos_spawn_air(LINDA_SPAWN_HEAT | LINDA_SPAWN_TOXINS | LINDA_SPAWN_OXYGEN, 5)
+
+/obj/effect/anomaly/pyro/detonate()
+	INVOKE_ASYNC(src, .proc/makepyroslime)
+
+/obj/effect/anomaly/pyro/proc/makepyroslime()
+	var/turf/simulated/T = get_turf(src)
+	if(istype(T))
+		T.atmos_spawn_air(LINDA_SPAWN_HEAT | LINDA_SPAWN_TOXINS | LINDA_SPAWN_OXYGEN, 500) //Make it hot and burny for the new slime
+	var/new_colour = pick("red", "orange")
+	var/mob/living/simple_animal/slime/S = new(T, new_colour)
+	S.rabid = TRUE
+	S.amount_grown = SLIME_EVOLUTION_THRESHOLD
+	S.Evolve()
+	var/datum/action/innate/slime/reproduce/A = new
+	A.Grant(S)
+
+	var/list/mob/dead/observer/candidates = SSghost_spawns.poll_candidates("Do you want to play as a pyroclastic anomaly slime?", ROLE_SENTIENT, FALSE, 100, source = S, role_cleanname = "pyroclastic anomaly slime")
+	if(LAZYLEN(candidates))
+		var/mob/dead/observer/chosen = pick(candidates)
+		S.key = chosen.key
+		S.mind.special_role = SPECIAL_ROLE_PYROCLASTIC_SLIME
+		log_game("[key_name(S.key)] was made into a slime by pyroclastic anomaly at [AREACOORD(T)].")
 
 /////////////////////
 
 /obj/effect/anomaly/bhole
 	name = "vortex anomaly"
 	icon_state = "bhole3"
 	desc = "That's a nice station you have there. It'd be a shame if something happened to it."
-
-/obj/effect/anomaly/bhole/New()
-	..()
-	aSignal.origin_tech = "engineering=7"
+	aSignal = /obj/item/assembly/signaler/anomaly/vortex
 
 /obj/effect/anomaly/bhole/anomalyEffect()
 	..()
 	if(!isturf(loc)) //blackhole cannot be contained inside anything. Weird stuff might happen
 		qdel(src)
 		return
 
-	grav(rand(0,3), rand(2,3), 50, 25)
+	grav(rand(0, 3), rand(2, 3), 50, 25)
 
 	//Throwing stuff around!
-	for(var/obj/O in orange(1,src))
+	for(var/obj/O in range(2, src))
```

Comment:
`atom_movable` to also throw mobs maybe?

Reply:
it does pull mobs in; this portion is design to throw the object *at* mobs.